jetty-util/src/main/java/org/eclipse/jetty/util/MutableIndex.java

```diff
@@ -0,0 +1,102 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public interface MutableIndex<V> extends Index<V>
```

Comment:
Again, rather than a whole new class, I'd much rather have an `immutable()` or `mutable()` (depending on default) method on the `Index.Builder`.  Just like the JVM collections, they all have the mutable APIs.

Reply:
Well, I'm very strongly in favor of separating the mutable vs the immutable API of an index as that makes the intent crystal clear when reading the code.

But having a single entry point to build both is the right thing to do IMO, so I've made `MutableIndex.Builder` not instanciatable and introduced a `Index.Builder.mutable()` method to create a mutable index.

Reply:
Can you at least move this to be a subinterface of Index, so it matches the pattern we have in HttpFields and HttpUri

Reply:
You mean a nested interface I suppose, because it already is a subinterface of Index?

Reply:
yep

Reply:
Done.